File: rides-service/src/main/java/io/simakkoi9/ridesservice/service/kafka/KafkaProducer.java
package io.simakkoi9.ridesservice.service.kafka;

import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class KafkaProducer {

    private final KafkaTemplate<String, List<Long>> kafkaTemplate;
    private final KafkaTemplate<String, String> ratingKafkaTemplate;

    public void sendDriverIdList(String rideId, List<Long> driverIdList) {
        kafkaTemplate.send("drivers-topic", rideId, driverIdList);
    }

    public void sendPersonId(String personId, String rate) {
        ratingKafkaTemplate.send("person-topic", personId, rate);
    }

}

File: driver-service/src/main/kotlin/io/simakkoi9/driverservice/service/kafka/KafkaProducer.kt
package io.simakkoi9.driverservice.service.kafka

import io.simakkoi9.driverservice.model.dto.kafka.KafkaDriverResponse
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class KafkaProducer(
    private val kafkaTemplate: KafkaTemplate<String, KafkaDriverResponse>
) {

    fun sendDriver(rideId: String, kafkaDriverResponse: KafkaDriverResponse) {
        kafkaTemplate.send("rides-topic", rideId, kafkaDriverResponse)
    }

}